services/data-info/src/data_info/routes/avus.clj

```diff
@@ -12,26 +13,29 @@
 
     (GET* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :return AVUGetResult
+      :return MetadataListing
+      :middlewares [wrap-metadata-base-url]
       :summary "List AVUs (administrative)"
-      :description (str "List iRODS AVUs associated with a data item. Include administrative/system AVUs."
-(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE"))
+      :description
+          (str "List all AVUs associated with a data item. Include administrative/system iRODS AVUs."
+(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE")
+(get-endpoint-delegate-block "metadata" "GET /avus/{target-type}/{target-id}"))
       (svc/trap uri meta/admin-metadata-get data-id))
 
-    (POST* "/avus" [:as {uri :uri}]
+    (PUT* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :body [{:keys [irods-avus]} (describe AVUListing "An list of AVUs to add")]
+      :body [{:keys [irods-avus]} (describe AVUListing "A list of AVUs to add")]
       :return AVUChangeResult
       :summary "Add AVUs (administrative)"
-      :description (str "Associate AVUs with a data item. Allow adding any AVU."
+      :description (str "Associate iRODS AVUs with a data item. Allow adding any AVU."
 (get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE, ERR_DOES_NOT_EXIST, ERR_NOT_WRITEABLE, ERR_NOT_AUTHORIZED"))
       (svc/trap uri meta/admin-metadata-add data-id irods-avus))
```

Comment:
The argument that `PUT` should be idempotent is the reason that I thought `PUT` should only add/update AVUs (or maybe it should be a `PATCH` endpoint instead?).

Metadata AVUs without IDs in the `set` request will always have a new ID after each subsequent request, since only AVUs with IDs are updated, and every other AVU will be deleted/added by the request, so it seemed like the `set` endpoint is not idempotent.

The highlighted section in the `PUT` quote from the RFC seems to actually argue for the opposite in this case, since the `set` endpoint will be modifying the existing _list_ of AVUs, by setting it to the list given in the request, whereas the `add/update` endpoint will only update or append to the list. I've also seen it mentioned elsewhere that a `PUT` request should create or replace, so it makes sense to me that the `set` endpoint should be a `PUT` request.

I'm fine with switching the data-info endpoints back, and then switching the metadata PUT vs POST endpoints to be consistent with data-info instead, unless anyone has better arguments to keep it this way.